src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length < 1) {
+      return '';
+    }
```

Comment:
we'll have an `empty string` as the result with an empty array as a parametr anyway, so we don't need this `if` statement